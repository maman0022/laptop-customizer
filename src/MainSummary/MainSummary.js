import React from 'react';
import SummaryItem from '../SummaryItem/SummaryItem'
import './MainSummary.css'

export default function MainSummary(props) {
  return (
    <section className="main__summary">
      <h2>Your cart</h2>
      <SummaryItem USCurrencyFormat={props.USCurrencyFormat} state={props.state}/>
      <div className="summary__total">
        <div className="summary__total__label">Total</div>
        <div className="summary__total__value">
          {props.USCurrencyFormat.format(props.total)}
        </div>
      </div>
    </section>
  )
}