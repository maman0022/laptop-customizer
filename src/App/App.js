import React, { Component } from 'react';
import store from '../STORE';
import Nav from '../Nav/Nav';
import MainForm from '../MainForm/MainForm';
import MainSummary from '../MainSummary/MainSummary';
import './App.css';

// This object will allow us to
// easily convert numbers into US dollar values
export const USCurrencyFormat = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD'
});

class App extends Component {
  state = {
    selected: {
      Processor: {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      },
      'Operating System': {
        name: 'Ubuntu Linux 16.04',
        cost: 200
      },
      'Video Card': {
        name: 'Toyota Corolla 1.5v',
        cost: 1150.98
      },
      Display: {
        name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
        cost: 1500
      }
    }
  };

  updateFeature = (feature, newValue) => {
    const selected = Object.assign({}, this.state.selected);
    selected[feature] = newValue;
    this.setState({
      selected
    });
  };

  render() {
    const total = Object.keys(this.state.selected).reduce(
      (acc, curr) => acc + this.state.selected[curr].cost,
      0
    );

    return (
      <div className="App">
        <Nav />
        <main>
          <MainForm USCurrencyFormat={USCurrencyFormat} state={this.state} updateFeature={this.updateFeature} store={store}/>
          <MainSummary USCurrencyFormat={USCurrencyFormat} total={total} state={this.state} />
        </main>
      </div>
    );
  }
}

export default App;