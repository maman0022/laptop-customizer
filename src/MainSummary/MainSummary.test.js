import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainSummary from './MainSummary';
import { USCurrencyFormat } from '../App/App';

let total = 1523.66;

describe('MainSummary Component', () => {
  it('the summary renders with a dummy total(1523.66) and reference to SummmaryItem component', () => {
    const wrapper = shallow(<MainSummary USCurrencyFormat = { USCurrencyFormat } total={total}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})