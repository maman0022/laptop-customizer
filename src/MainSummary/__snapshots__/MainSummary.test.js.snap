// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSummary Component the summary renders with a dummy total(1523.66) and reference to SummmaryItem component 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <SummaryItem
    USCurrencyFormat={NumberFormat {}}
  />
  <div
    className="summary__total"
  >
    <div
      className="summary__total__label"
    >
      Total
    </div>
    <div
      className="summary__total__value"
    >
      $1,523.66
    </div>
  </div>
</section>
`;
